### PR TITLE
return status from the event publish API

### PR DESCRIPTION
**Pull request by nickmahilani**
### Context

Return PublishStatus from the event publish API to indicate status to caller

### Checklist

- [X] `./gradlew build` compiles code correctly
- [X] Added new tests where applicable
- [X] `./gradlew test` passes all tests
- [X] Extended README or added javadocs where applicable
- [X] Added copyright headers for new files from `CONTRIBUTING.md`

### CODE REVIEW

File: mantis-publish-core/src/main/java/io/mantisrx/publish/api/PublishStatus.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    SKIPPED_NO_SUBSCRIPTIONS(Status.PRECONDITION_FAILED),
+    /**
+     * Event was not enqueued since the Mantis Publish client is not enabled in configuration
+     */
+    SKIPPED_CLIENT_NOT_ENABLED(Status.PRECONDITION_FAILED),
+
+    /**
+     * Event was not enqueued since the event was invalid (either null or empty)
+     */
+    SKIPPED_INVALID_EVENT(Status.PRECONDITION_FAILED),
```
Comment (neerajrj) [inline mantis-publish-core/src/main/java/io/mantisrx/publish/api/PublishStatus.java]: Does this also include the case the serialization failed before sending the event?

Comment (nickmahilani) [reply mantis-publish-core/src/main/java/io/mantisrx/publish/api/PublishStatus.java]: That would be a `FAILED_SERIALIZATION(Status.Failed)` we could add later once we propagate `Promise<PublishStatus>` along with the event all the way down to the transport

Comment (neerajrj) [reply mantis-publish-core/src/main/java/io/mantisrx/publish/api/PublishStatus.java]: Oh I don't see that enum in PublishStatus.java?

Comment (nickmahilani) [reply mantis-publish-core/src/main/java/io/mantisrx/publish/api/PublishStatus.java]: we can add more Status fields to the enum later when we implement the async propagation. I got half way implementing the async prop and realized the netty side of changes will be non-trivial so backtracked and scoped this to sync status updates for now.. we can add those in when we have the full impl.. This in its current form is useful already for some Netflix users.

Comment (neerajrj) [reply mantis-publish-core/src/main/java/io/mantisrx/publish/api/PublishStatus.java]: sounds reasonable.

**Pull request merged**